src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -295,25 +388,18 @@ public override RunVersionOne VisitRunVersionOne(RunVersionOne node)
                 {
                     run.Tags.UnionWith(node.Tags);
                 }
-
-                SarifLog.Runs.Add(run);
-
-                VisitToolVersionOne(node.Tool);
             }
 
             return null;
         }
 
-        public override ResultVersionOne VisitResultVersionOne(ResultVersionOne node)
+        public Tool TransformToolVersionOne(ToolVersionOne node)
```

Comment:
@EasyRhin0 @michaelcfanning Ok, I see what you're doing here. Rather than overriding the visitor-provided tool-visiting method, you provide your own, which returns the type you need (`Tool`) rather than `ToolVersionOne`. And I see you did that in your `TransformInvocationOne` method above, as well. This all makes sense.

From these two examples, we gain an insight: we don't need to base our code on the auto-generated v1 visitor at all! It  would be a small change to your existing code to create a class that accepts a v1 OM, and walks down it using methods that return the appropriate v2 objects.

Remember, the reason we code-gen'd the visitor classes was so that the visitors would always be up to date when the schema changed. But the v1 schema is frozen! So we can hard-code our converter class to the existing, frozen schema.

Using the code-gen'd visitor was a good way to get started, but now that you've demonstrated a better way to code this, we should go all in on it.

By the way, remember at the very beginning when you recognized that the code-gen'd visitor returned a v1 log object -- and so you had to hack around it by synthesizing a v2 log object in a property, and then pulling out the property at the end? This small redesign would avoid having to do that.

Reply:
Perfect

Reply:
THIS will be the next PR :)